outbound/state/tests/test_outbound_state.py

```diff
@@ -0,0 +1,169 @@
+                                              sequence_number=2),
+                            MessageHeader(sequence_number=3),
+                            MessageHeader(sequence_number=7),
+                            ReferenceTransactionNumber(reference=None),
+                            ReferenceTransactionType(ReferenceTransactionType.TransactionType.ACCEPTANCE)]
+                state.create_new_outbound_state(segments, OPERATION_ID)
+        with self.subTest('None timestamp'):
+            with self.assertRaises(ValueError):
+                segments = [InterchangeHeader(sender='test_sender',
+                                              recipient='test_recipient',
+                                              date_time=None,
+                                              sequence_number=2),
+                            MessageHeader(sequence_number=3),
+                            MessageHeader(sequence_number=7),
+                            ReferenceTransactionNumber(reference=1),
+                            ReferenceTransactionType(ReferenceTransactionType.TransactionType.ACCEPTANCE)]
+                state.create_new_outbound_state(segments, OPERATION_ID)
+        with self.subTest('Empty timestamp string'):
+            with self.assertRaises(ValueError):
+                segments = [InterchangeHeader(sender='test_sender',
+                                              recipient='test_recipient',
+                                              date_time='',
+                                              sequence_number=2),
+                            MessageHeader(sequence_number=3),
+                            MessageHeader(sequence_number=7),
+                            ReferenceTransactionNumber(reference=1),
+                            ReferenceTransactionType(ReferenceTransactionType.TransactionType.ACCEPTANCE)]
+                state.create_new_outbound_state(segments, OPERATION_ID)
+        with self.subTest('None transaction_type'):
+            with self.assertRaises(AttributeError):
+                segments = [InterchangeHeader(sender='test_sender',
+                                              recipient='test_recipient',
+                                              date_time='12:00',
+                                              sequence_number=2),
+                            MessageHeader(sequence_number=3),
+                            MessageHeader(sequence_number=7),
+                            ReferenceTransactionNumber(reference=1),
+                            ReferenceTransactionType(None)]
+                state.create_new_outbound_state(segments, OPERATION_ID)
+        with self.subTest('Not existing transaction_type'):
+            with self.assertRaises(AttributeError):
+                segments = [InterchangeHeader(sender='test_sender',
+                                              recipient='test_recipient',
+                                              date_time='12:00',
```

Comment:
In IDE warning is raised, Expected type 'datetime', got 'str' instead. InterchangeHeader is expecting datetime to be passed in parameters

Reply:
it does not impact the test result, but I can wrap it into the datetime object

Reply:
done!